subsys/bluetooth/host/conn.c

```diff
@@ -513,6 +546,11 @@ static bool send_frag(struct bt_conn *conn, struct net_buf *buf, uint8_t flags,
 	if (IS_ENABLED(CONFIG_BT_ISO) && conn->type == BT_CONN_TYPE_ISO) {
 		err = send_iso(conn, buf, flags);
 	} else if (IS_ENABLED(CONFIG_BT_CONN)) {
+		/* Parts of buffer user data may be overridden by send_acl().
+		 * Set it to NULL here and reset in case of failure to ensure
+		 * the user data is not interpreted as an unaligned pointer.
+		 */
+		tx_data(buf)->tx = NULL;
 		err = send_acl(conn, buf, flags);
```

Comment:
Something is fishy here. I don't fully know what is happening here, but clearly `send_acl` expects to have control over `buf->user_data` and it's not expecting it to be a `tx_meta`. If `send_acl` requires the user data buffer to be zeroed out, then it's safer to memset the whole struct here. (In case of future addition of members and padding.) The whole reuse seems unsafe to me. But I understand that this was like this before?

Reply:
The user data is used in multiple places and there are multiple different structs overlaid this buffer. This specific problem was introduced in this PR because we later on interpret the user data as a pointer.